### PR TITLE
ENH multi-extension&symlink non-editable for files + cache score computations

### PR DESCRIPTION
**Pull request by tomMoral**
This PR back-port the fixes I had to make to make the server run:
* Use a make develop
* Allow for multi-extension
* Cache score computations

I also included a fix to allow access to code in collab phase.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,6 +15,12 @@ install:
 	cd ramp-engine && pip install . && cd ..
 	cd ramp-utils && pip install . && cd ..
 
+develop:
+	cd ramp-frontend && pip install -e . && cd ..
+	cd ramp-database && pip install -e . && cd ..
+	cd ramp-engine && pip install -e . && cd ..
+	cd ramp-utils && pip install -e . && cd ..
```
Comment (rth) [inline Makefile]: Thant's identical to `inplace` below..

File: ramp-database/ramp_database/tools/submission.py [hunk 2]
```diff
@@ -68,7 +69,7 @@ def add_submission(session, event_name, team_name, submission_name,
     submission = (session.query(Submission)
                          .filter(Submission.name == submission_name)
                          .filter(Submission.event_team == event_team)
-                         .one_or_none())
+                         .first())
```
Comment (rth) [inline ramp-database/ramp_database/tools/submission.py]: I would rather keep `one_or_none` then explicitly handle the None case. Otherwise .first() would return an empty list, if none match, right?

Actually, `None` is already handled below, why is this necessary then?

File: ramp-database/ramp_database/tools/submission.py [hunk 3]
```diff
@@ -415,18 +425,26 @@ def get_time(session, submission_id):
     computation_time : pd.DataFrame
         A pandas dataframe containing the computation time of each fold.
     """
-    results = defaultdict(list)
-    all_cv_folds = (session.query(SubmissionOnCVFold)
+    cache_filename = os.path.join(
+            CACHE_DIR, 'times_submission_{:09}.pkl.zip'
+            .format(submission_id))
+    if os.path.exists(cache_filename):
+        times = pd.read_pickle(cache_filename)
```
Comment (rth) [inline ramp-database/ramp_database/tools/submission.py]: Not really sure what is going on here, could you explain? (and also if some of this wasn't addressed in some of the other PRs).

File: ramp-database/ramp_database/tools/submission.py [hunk 4]
```diff
@@ -444,21 +462,28 @@ def get_scores(session, submission_id):
     scores : pd.DataFrame
         A pandas dataframe containing the scores of each fold.
     """
-    results = defaultdict(list)
-    index = []
-    all_cv_folds = (session.query(SubmissionOnCVFold)
+    cache_filename = os.path.join(
+            CACHE_DIR, 'scores_submission_{:09}.pkl.zip'
```
Comment (rth) [inline ramp-database/ramp_database/tools/submission.py]: Same as above

File: ramp-frontend/ramp_frontend/views/ramp.py [hunk 5]
```diff
@@ -823,14 +823,20 @@ def view_model(submission_hash, f_name):
                         submission, filename)
             )
 
+            workflow_element = WorkflowElement.query.filter_by(
+                name=filename.split('.')[0], workflow=event.workflow).one()
```
Comment (rth) [inline ramp-frontend/ramp_frontend/views/ramp.py]: Maybe define a separate helper function to get root ° extension from a path, and use it consistently? `os.path.splitext` is not doing what we need?

**Comment by agramfort** [comment]
@tomMoral can you see @rth comment's before we merge this?

File: ramp-database/ramp_database/tools/submission.py [hunk 6]
```diff
@@ -32,6 +32,7 @@
 from ._query import select_submission_file_type_by_name
 from ._query import select_team_by_name
 
+CACHE_DIR = '/home/ramp/ramp_deployment/cache/'
```
Comment (kegl) [inline ramp-database/ramp_database/tools/submission.py]: I think this should not be hardcoded here, rather handled using config.

Comment (glemaitre) [reply ramp-database/ramp_database/tools/submission.py]: We used sqlalchemy to differ the loading. We don't need caching anymore.
